analyzers/src/SonarAnalyzer.Common/Helpers/TypeHelper.cs

```diff
@@ -40,7 +45,18 @@ internal static class TypeHelper
             self?.TypeKind == typeKind;
 
         public static bool IsNullableValueType(this ITypeSymbol self) =>
-            self.IsStruct() && self.OriginalDefinition.Is(KnownType.System_Nullable_T);
+            self.IsStruct() && self is { SpecialType: SpecialType.System_Nullable_T } or { OriginalDefinition.SpecialType: SpecialType.System_Nullable_T };
+
+        public static bool IsNonNullableValueType(this ITypeSymbol self) =>
+            self.IsStruct() && !self.IsNullableValueType();
+
+        public static bool IsEnum(this ITypeSymbol self) =>
```

Comment:
This seems to be added and never used. As it's not needed (yet), I'd add it later when we really need it (the check change).